service/src/main/java/bio/terra/landingzone/library/landingzones/definition/factories/CromwellBaseResourcesFactory.java

```diff
@@ -274,11 +279,28 @@ public Deployable definition(DefinitionContext definitionContext) {
               .withSystemAssignedManagedServiceIdentity()
               .defineAgentPool(
                   nameGenerator.nextName(ResourceNameGenerator.MAX_AKS_AGENT_POOL_NAME_LENGTH))
-              .withVirtualMachineSize(ContainerServiceVMSizeTypes.STANDARD_A2_V2)
-              .withAgentPoolVirtualMachineCount(1)
-              .withAgentPoolMode(
-                  AgentPoolMode.SYSTEM) // TODO VM Size? Pool Machine count? AgentPoolMode?
-              .withVirtualNetwork(vNetwork.id(), Subnet.AKS_SUBNET.name())
+              .withVirtualMachineSize(
+                  ContainerServiceVMSizeTypes.fromString(
+                      parametersResolver.getValue(ParametersNames.AKS_MACHINE_TYPE.name())))
+              .withAgentPoolVirtualMachineCount(
+                  Integer.parseInt(
+                      parametersResolver.getValue(ParametersNames.AKS_NODE_COUNT.name())))
+              .withAgentPoolMode(AgentPoolMode.SYSTEM)
+              .withVirtualNetwork(vNetwork.id(), Subnet.AKS_SUBNET.name());
+
+      if (Boolean.getBoolean(
+          parametersResolver.getValue(ParametersNames.AKS_AUTOSCALING_ENABLED.name()))) {
+        int min =
+            Integer.parseInt(
+                parametersResolver.getValue(ParametersNames.AKS_AUTOSCALING_MIN.name()));
```

Comment:
To avoid hard to the debug issues, adding validation for ranges and values of these parameters would be helpful.